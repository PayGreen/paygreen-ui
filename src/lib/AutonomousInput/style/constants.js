import { math } from 'polished';

const fieldHeight = '34px';
const fieldRadius = math(fieldHeight + '/2');

export {
    fieldHeight,
    fieldRadius
};