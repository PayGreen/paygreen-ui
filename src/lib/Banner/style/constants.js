const color = {
    first: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.primary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientBase']
    },
    second: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.secondary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientShade']
    }
};

export {
    color
};