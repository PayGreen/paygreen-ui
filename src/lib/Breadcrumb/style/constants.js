import { transparentize } from 'polished';

const textColor = {
    default: {
        original: props => props.theme.wab.grey30,
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    link: {
        original: props => transparentize(0.3, props.theme.color[props.colorTheme].main),
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    hover: {
        original: props => props.theme.color[props.colorTheme].main,
        reverse: props => transparentize(0.05, props.theme.wab.white00),
    }
};

export {
    textColor
};