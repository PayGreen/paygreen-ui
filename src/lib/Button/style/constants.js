const colorCollection = {
    main: props => props.theme.color[props.colorTheme].main,
    white: props => props.theme.wab.white00,
    lightGrey: props => props.theme.wab.white20,
    grey: props => props.theme.wab.grey30,
};

const buttonColors = {
    text: {
        fill: {
            original: {
                enabled: colorCollection.white,
                disabled: colorCollection.lightGrey,
            },
            reverse: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
        },
        line: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
    },
    bg: {
        fill: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
        line: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
    },
};

const enableType = props => (props.isDisabled ? 'disabled' : 'enabled');

const opacity = {
    original: 0.5,
    reverse: 0.6,
};

export { buttonColors, enableType, opacity };
