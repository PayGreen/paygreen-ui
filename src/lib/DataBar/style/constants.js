import { math } from 'polished';

const mainColor = {
    theme: props => props.theme.color[props.colorTheme].main,
    status: props => props.theme.color.status[props.colorStatus].main,
};

const barRadius = props => math(props.theme.dataBar.height + '/2');

export { barRadius, mainColor };
