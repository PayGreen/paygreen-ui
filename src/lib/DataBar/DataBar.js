import React from 'react';
import PropTypes from 'prop-types';
import {
    blockWidthOptions,
    blockWidthDefault,
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    formStatusOptions,
    fontSizeOptions,
    fontSizeDefault,
    greyOptions,
} from '../../shared/constants';
import { DataBarBase, Bar, BarSection } from './style';
import Text from '../Text/Text';
import DataLegend from '../DataLegend/DataLegend';

const DataBar = props => {
    const {
        blockWidth,
        data,
        hasDataLegend,
        minValue,
        maxValue,
        unit,
        hasBackground,
        ...rest
    } = props;

    // to calculate the value depending on unit type and to force calculation to percent if needed
    const calcValue = (value, toPercent) => {
        let percent = ((value - minValue) / (maxValue - minValue)) * 100;
        if (!toPercent) {
            return unit.length && unit !== '%' ? value : percent.toFixed(2);
        } else return percent.toFixed(2);
    };

    return (
        <DataBarBase {...rest}>
            <Bar
                {...rest}
                hasBackground={hasBackground}
                blockWidth={blockWidth}
            >
                {data.map((dataItem, index) => (
                    <BarSection
                        {...rest}
                        key={index}
                        dataWidth={calcValue(dataItem.value, true)}
                        colorStatus={dataItem.colorStatus}
                        colorTheme={dataItem.colorTheme}
                    />
                ))}
            </Bar>

            {hasDataLegend
                ? data.map((dataItem, index) => (
                      <DataLegend
                          {...rest}
                          key={index}
                          colorStatus={dataItem.colorStatus}
                          colorTheme={dataItem.colorTheme}
                          unit={unit.length ? unit : '%'}
                          value={calcValue(dataItem.value, false)}
                      >
                          {dataItem.legend && dataItem.legend.length ? (
                              <Text
                                  {...rest}
                                  colorPallet={colorPalletOptions.wab}
                                  htmlTag="span"
                              >
                                  {dataItem.legend}
                              </Text>
                          ) : null}
                      </DataLegend>
                  ))
                : null}
        </DataBarBase>
    );
};

DataBar.propTypes = {
    blockWidth: PropTypes.oneOf(Object.values(blockWidthOptions)),
    colorPallet: PropTypes.oneOf([
        colorPalletOptions.theme,
        colorPalletOptions.status,
    ]),
    data: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.number.isRequired,
            legend: PropTypes.string,
            colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions))
                .isRequired,
            colorStatus: PropTypes.oneOf(Object.values(formStatusOptions))
                .isRequired,
        }),
    ).isRequired,
    hasDataLegend: PropTypes.bool,
    minValue: PropTypes.number,
    maxValue: PropTypes.number,
    textSize: PropTypes.oneOf(Object.values(fontSizeOptions)),
    unit: PropTypes.string,
    hasBackground: PropTypes.bool,
    backgroundColor: PropTypes.oneOf(Object.values(greyOptions)),
};

DataBar.defaultProps = {
    blockWidth: blockWidthDefault,
    colorPallet: colorPalletDefault,
    hasDataLegend: false,
    minValue: 0,
    maxValue: 100,
    textSize: fontSizeDefault,
    unit: '%',
    hasBackground: false,
    backgroundColor: greyOptions.grey10,
};

export default DataBar;
