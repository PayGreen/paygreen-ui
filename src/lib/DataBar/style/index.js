import styled from 'styled-components';
import { zeroValueStyle, valueStyle, hasBackgroundStyle } from './base';
import { barRadius } from './constants';

const Bar = styled.div`
    max-width: ${props => props.theme.dataBar.width[props.blockWidth]};
    height: ${props => props.theme.dataBar.height};
    display: flex;
    border-radius: ${barRadius};
    overflow: hidden;

    ${props => (props.hasBackground ? hasBackgroundStyle : null)};
`;

const BarSection = styled.div`
    margin-right: ${props => props.theme.line};
    width: ${props => props.dataWidth + '%'};
    min-width: ${props => props.theme.dataBar.height};

    &:last-of-type {
        margin-right: 0;
        border-top-right-radius: ${barRadius};
        border-bottom-right-radius: ${barRadius};
    }

    ${props => (props.dataWidth > 0 ? valueStyle : zeroValueStyle)};
`;

export { BarSection, Bar };
