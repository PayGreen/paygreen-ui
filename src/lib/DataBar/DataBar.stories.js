import React from 'react';

import {
    folder,
    spaceOptions,
    colorThemeOptions,
    colorThemeDefault,
    colorPalletOptions,
    colorPalletDefault,
    formStatusOptions,
    formStatusDefault,
    fontSizeOptions,
    fontSizeDefault,
    greyOptions,
} from '../../shared/constants';
import labels from '../../shared/labels';
import DataBar from './DataBar';

const defaultValue = 25;
const valueOptions = {
    range: true,
    max: 100,
};

const { wab, ...barColorPalletOptions } = colorPalletOptions;

export default {
    title: folder.graph + 'Databar',
    argTypes: {
        hasBackground: {
            name: labels.hasBackground,
            control: 'boolean',
        },
        backgroundColor: {
            name: labels.backgroundColor,
            options: Object.values(greyOptions),
            control: 'select',
        },
        textSize: {
            name: labels.textSize,
            control: 'select',
            options: Object.values(fontSizeOptions),
        },
        unit: {
            name: labels.unity,
            control: 'text',
        },
        colorPallet: {
            name: labels.colorPallet,
            options: Object.values(colorPalletOptions),
            control: 'radio',
        },
        maxValue: {
            name: labels.maxValue,
            control: 'number',
        },
        value: {
            name: labels.value,
            control: 'range',
        },
        value2: {
            name: labels.value + '2',
            control: 'range',
        },
        value3: {
            name: labels.value + '3',
            control: 'range',
        },
        colorTheme: {
            name: labels.colorTheme,
            options: Object.values(colorThemeOptions),
            control: 'select',
        },
        colorStatus: {
            name: labels.colorStatus,
            options: Object.values(formStatusOptions),
            control: 'select',
        },
        blockWidth: {
            name: labels.blockWidth,
            options: Object.values(spaceOptions),
            control: 'select',
        },
    },
};

export const DataBarClassic = ({ value, colorTheme, colorStatus, ...args }) => (
    <DataBar
        {...args}
        data={[
            {
                value: value,
                colorTheme: colorTheme,
                colorStatus: colorStatus,
            },
        ]}
    />
);

export const DataBarMulti = ({ value, value2, value3, ...args }) => (
    <DataBar
        {...args}
        data={[
            {
                value: value,
            },
            {
                value: value2,
                colorTheme: colorThemeOptions.tertiary,
                colorStatus: formStatusOptions.warning,
            },
            {
                value: value3,
                colorTheme: colorThemeOptions.quaternary,
                colorStatus: formStatusOptions.danger,
            },
        ]}
    />
);

DataBarClassic.args = {
    hasBackground: true,
    backgroundColor: greyOptions.grey10,
    textSize: fontSizeDefault,
    unit: '%',
    colorPallet: colorPalletDefault,
    maxValue: 100,
    value: defaultValue,
    colorTheme: colorThemeDefault,
    colorStatus: formStatusDefault,
    blockWidth: spaceOptions.md,
};

DataBarMulti.args = {
    hasBackground: true,
    colorPallet: colorPalletDefault,
    blockWidth: spaceOptions.md,
    value: defaultValue,
    value2: defaultValue,
    value3: defaultValue,
};
