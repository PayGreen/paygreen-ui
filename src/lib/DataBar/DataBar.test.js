import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import DataBar from './DataBar';
import { colorThemeDefault, formStatusDefault } from '../../shared/constants';

const data = [
    {
        currentValue: 20,
        legend: 'lÃ©gende de la barre',
        colorStatus: formStatusDefault,
        colorTheme: colorThemeDefault,
    },
];

it('renders without crashing', () => {
    const dataBar = TestRenderer.create(
        <DataBar theme={ThemeDefault} data={data} />,
    );
    expect(dataBar.toJSON()).toMatchSnapshot();
});
