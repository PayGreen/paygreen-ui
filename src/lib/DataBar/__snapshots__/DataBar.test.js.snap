// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="sc-AxjAm gmXMlt"
  >
    <div
      className="sc-AxirZ kImrdU"
    />
  </div>
</div>
`;
